cmd/mobynit/main.go

```diff
@@ -65,46 +74,68 @@ func mountContainer(containerID, graphDriver string) string {
 	return newRootPath
 }
 
-func main() {
-	// Any mounts done by initrd will be transfered in the new root
-	mounts, err := mount.GetMounts()
-
-	rawGraphDriver, err := ioutil.ReadFile("/current/boot/storage-driver")
+func getStorageDriverAndContainerID(sysroot string) (string, string) {
+	rawGraphDriver, err := ioutil.ReadFile(filepath.Join(sysroot, "/current/boot/storage-driver"))
 	if err != nil {
 		log.Fatal("could not get storage driver:", err)
 	}
 	graphDriver := strings.TrimSpace(string(rawGraphDriver))
 
-	current, err := os.Readlink("/current")
+	current, err := os.Readlink(filepath.Join(sysroot, "/current"))
 	if err != nil {
 		log.Fatal("could not get container ID:", err)
 	}
 	containerID := filepath.Base(current)
 
-	if err := unix.Mount("", "/", "", unix.MS_REMOUNT, ""); err != nil {
-		log.Fatal("error remounting root as read/write:", err)
+	return graphDriver, containerID
+}
+
+func main() {
+	sysrootPtr := flag.String("sysroot", "", "root of partition e.g. /mnt/sysroot/inactive. Mount destination is returned in stdout")
+	flag.Parse()
+
+	// Any mounts done by initrd will be transfered in the new root
+	mounts, err := mount.GetMounts()
+	if err != nil {
+		log.Fatal("could not get mounts:", err)
 	}
 
-	newRoot := mountContainer(containerID, graphDriver)
+	var graphDriver, containerID string
+
+	// If a custom sysroot is passed, use it instead of LAYER_ROOT
+	if *sysrootPtr != "" {
+		graphDriver, containerID = getStorageDriverAndContainerID(*sysrootPtr)
+		newRootPath := mountContainer(filepath.Join(*sysrootPtr, LAYER_ROOT), containerID, graphDriver)
+		fmt.Print(newRootPath)
+
+	} else {
+		graphDriver, containerID = getStorageDriverAndContainerID("")
 
-	for _, mount := range mounts {
-		if mount.Mountpoint == "/" {
-			continue
+		if err := unix.Mount("", "/", "", unix.MS_REMOUNT, ""); err != nil {
+			log.Fatal("error remounting root as read/write:", err)
 		}
-		if err := unix.Mount(mount.Mountpoint, filepath.Join(newRoot, mount.Mountpoint), "", unix.MS_MOVE, ""); err != nil {
-			log.Println("could not move mountpoint:", mount.Mountpoint, err)
+
+		newRoot := prepareForPivot("", containerID, graphDriver)
+
+		for _, mount := range mounts {
```

Comment:
Isn't this also preparing it for pivot? is there any reason not to move it in `prepareForPivot`?

Reply:
The mounts variable is in this scope. Possibly can pass it in `prepareForPivot`.
My aim to reduce the diff to a small commit that just does the bare minimum seems to have turned into a larger commit.